ocaml/xapi/xapi.ml

```diff
@@ -159,11 +159,18 @@ let random_setup () =
   finally (fun () -> really_input chan s 0 n) (fun () -> close_in chan) ;
   Random.full_init (Array.init n (fun i -> Char.code (Bytes.get s i)))
 
+let dummy_fd, dummy_fd2 = Unix.pipe ()
+
+let () = Unix.close dummy_fd
```

Comment:
This slots into the origin type of `ocaml/xapi/context.ml`, which needs to be handled by all of its user, like the `preauth`, `_client_of_origin`,  `parent_of_origin`, and others.

This needs a closer look, I doubt it can work as-is